test/Sema/noimplicitcopy_attr.swift

```diff
@@ -83,7 +83,7 @@ class MyGenericClass<T> {
         return nil
     }
 
-    func foo(@_noImplicitCopy _ t: T) { // expected-error {{@_noImplicitCopy may only be used on 'var' declarations}}
+    func foo(@_noImplicitCopy _ t: T) {
```

Comment:
NOTE: The error for using this on a generic parameter is a SILGen error, not a semi error, so it is in the SILGen test that I added. This is b/c the type checker doesn't have enough information to distinguish in between a type that contains a generic type and a type that just has a type parameter (consider UnsafePointer<T>, a loadable type at the SIL level despite the type parameter at the AST level).